### DISCUSSION TITLE
关于引入第三方npm包

### DISCUSSION THREAD

**Discussion by w990752800** [opentiny/tiny-engine#379] [Q&A]
我不想使用 cdn 形式，能在直接在代码中添加npm包，类似华为的vue的内置组件包。然后定义物料json, 直接使用而不用取配置json中的npm那些

**Comment by w990752800** [comment]
我记得看到一个说在 design-core 还是 canvas 中直接npm install 包的issues回答，为了解决cdn连接有问题的情况，怎么找不到了

**Reply by chilingling** [reply]
@w990752800 
内置的 tiny-vue 组件是特殊情况，如果不在物料 json 配置 npm 的 cdn 链接，则需要解决画布 & 预览 & 出码DSL 依赖声明、package.json 依赖声明等等场景。所以建议在物料 json 中配置 npm字段，避免强耦合的情况发生

**Reply by w990752800** [reply]
有么有简单的方案 ，让我自己下载依赖

**Reply by w990752800** [reply]
我们不让用cdn链接

**Reply by rhlin** [reply]
可以参考pr#360 改造一下脚本可以从npm包一起打包到设计器里， 可以绕开使用cdn场景。
​
如果你想内置到画布里， 你也可以在物料里不配置地址， 把组件配置到importmap里，但是出码等多个地方可能要联动改，建议还是配置一下地址。另外，后面物料要和画布解耦，包括tiny-vue组件的特殊情况都要解耦，不建议现阶段继续耦合在画布里。

**Reply by chilingling** [reply]
https://github.com/mjackson/unpkg  
可以自建 unpkg 源的，不需要用外部的 unpkg 源
@w990752800
